ad20e5eb15d037a31e3b484e9f57542dc927c0a9
http://sourceware.org/ml/gdb-cvs/2010-10/msg00005.html

### src/gdb/testsuite/ChangeLog	2010/09/30 18:58:07	1.2462
### src/gdb/testsuite/ChangeLog	2010/10/01 17:03:49	1.2463
## -1,3 +1,19 @@
+2010-10-01  Doug Evans  <dje@google.com>
+
+	* lib/gdb-python.exp: New file.
+	* gdb.python/py-block.exp: Use it.
+	* gdb.python/py-breakpoint.exp: Ditto.
+	* gdb.python/py-frame.exp: Ditto.
+	* gdb.python/py-inferior.exp: Ditto.
+	* gdb.python/py-param.exp: Ditto.
+	* gdb.python/py-prettyprint.exp: Ditto.
+	* gdb.python/py-shared.exp: Ditto.
+	* gdb.python/py-symbol.exp: Ditto.
+	* gdb.python/py-symtab.exp: Ditto.
+	* gdb.python/py-type.exp: Ditto.
+	* gdb.python/py-value.exp: Ditto.
+	* gdb.python/python.exp: Ditto.
+
 2010-09-30  Tom Tromey  <tromey@redhat.com>
 
 	* gdb.base/anon.exp: New file.
--- src/gdb/testsuite/lib/gdb-python.exp
+++ src/gdb/testsuite/lib/gdb-python.exp	2010-10-12 17:58:35.904468000 +0000
@@ -0,0 +1,47 @@
+# Copyright 2010 Free Software Foundation, Inc.
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Utilities for python-scripting related tests.
+
+# Run a command in GDB, and report a failure if a Python exception is thrown.
+# If report_pass is true, report a pass if no exception is thrown.
+
+proc gdb_py_test_silent_cmd { cmd name report_pass } {
+    global gdb_prompt
+
+    gdb_test_multiple $cmd $name {
+	-re "Traceback.*$gdb_prompt $" { fail $name }
+	-re "$gdb_prompt $"            { if $report_pass { pass $name } }
+    }
+}
+
+# Usage: gdb_py_test_multiple NAME INPUT RESULT {INPUT RESULT}...
+# Run a test named NAME, consisting of multiple lines of input.
+# After each input line INPUT, search for result line RESULT.
+# Succeed if all results are seen; fail otherwise.
+
+proc gdb_py_test_multiple { name args } {
+    global gdb_prompt
+    foreach {input result} $args {
+	if {[gdb_test_multiple $input "$name - $input" {
+	    -re "\[\r\n\]*($result)\[\r\n\]+($gdb_prompt | *>)$" {
+		pass "$name - $input"
+	    }
+	}]} {
+	    return 1
+	}
+    }
+    return 0
+}
